plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHiveFileSystem.java

```diff
@@ -488,6 +488,21 @@ public void testFileIteratorListing()
         assertEqualsIgnoreOrder(shallowListing, ImmutableList.of(baseFile));
     }
 
+    @Test
+    public void testDirectoryWithTrailingSpace()
+            throws Exception
+    {
+        Path basePath = new Path(getBasePath(), randomUUID().toString());
+        FileSystem fs = hdfsEnvironment.getFileSystem(TESTING_CONTEXT, basePath);
+        assertFalse(fs.exists(basePath));
+
+        Path path = new Path(basePath, "dir_with_space /foo.txt");
+        assertTrue(fs.createNewFile(path));
```

Comment:
I would recommend opening the file to cover the normal scenario. The existing flow (create, check, drop) without reading is rare in my opinion. 

The reason I'm suggesting this is `AzureBlobFileSystem.open` behavior is different from others.  https://issues.apache.org/jira/browse/HADOOP-18580